Java/syntax_test_java.java

```diff
@@ -312,3 +371,502 @@ void test2() {
 //            ^ support.class
     }
 }
+
+public enum FooEnum {
+//     ^^^^ storage.type.java
+  FOO;
+//^^^ constant.other.enum
+}
+// <- meta.class.java meta.class.body.java punctuation.section.class.end.java
```

Comment:
It makes sense, but I have a concern.
I have a plugin that takes the current cursor position and gives you the method and class name that the cursor is in. It does this by looking at all `meta.method` scopes and seeing if the cursor is in one of the scopes. It then looks for `entity.name.function` in the matching scope. It then does the same thing for the class and class name.

If `meta.method` only contains `entity.name.function` and not the `meta.method.body`, then I'm not sure how to find out the method name for a specific `meta.method.body`.